1-mandatory/2-people-i-know.js

```diff
@@ -374,26 +380,44 @@ const friends = [
 /*
 2) Aged 35 or Older
 In the above object you can see my friends and the colleagues of my friends.
-First, I want you to find all of my friends who are 35 or older.
+First, I want you to find all of my ----> friends who are 35 or older.
 */
 
-let thirtyFiveOrOlder = [];
+let thirtyFiveOrOlder = friends.filter((singleFriend) => singleFriend.age > 35);
 
 /*
 3) Find the email address
 Next, I want you to find all of my friends who work for "POWERNET" and then store their emails in the array below
 */
 
-let powerNetEmails = [];
+let powerNetFriends = friends.filter(
+  (singleFriend) => singleFriend.company === "POWERNET"
+);
+let powerNetEmails = powerNetFriends.map((singleFriend) => singleFriend.email);
 
 /*
 4) colleagues with "Stacie Villarreal"
-Next, I want you to find all of my friends who are colleagues of Stacie Villarreal.
+Next, I want you to find all of my --> friends who are colleagues of Stacie Villarreal.
 You can see who people's colleagues are by seeing the "colleagues" array in each of my friends objects.
 This time, I only want the full names ("<firstname> <lastname>") of my friends who are colleagues of hers.
 */
 
-let friendsWhoAreColleaguesOfStacie = [];
+// friends array
+//  -> person object
+//     -> name object
+//     -> colleagues array of objects
+//        -> skills array
+
+///////////////////// HELP!! ///////////////////
+let colleaguesOfStacie = friends.filter(
+  (singleFriend) => singleFriend.colleagues.includes("Stacie Villarreal") // this return true or false - not sure how to proceed from here
```

Comment:
this is all really good! nicely laid out, nice names of variables - good work :) made it really easy to read and check!

the issue here is you're filtering friends - going through and checking if each 'singleFriend.colleagues'  includes 'Stacie Villarreal', the colleagues array contains objects 😅  so you're looking at an array of the colleagues eg `{
        name: "Nannie Reyes",
        age: 47,
        skills: ["Sharing", "management", "Time"],
      }` and using that callback function `(singleFriend) => singleFriend.colleagues.includes("Stacie Villarreal")` to check if any of the colleagues objects are equal to the string 'Stacie Villarreal' but they obviously arent because you're comparing the object and string! if you changed it to 
`(singleFriend) => singleFriend.colleagues.map(colleague->colleague.name).includes("Stacie Villarreal")`
then that should work, does that make sense?

Reply:
Thanks for checking this out!
I'm not sure I understand the '->' symbol in (colleague**->**colleague.name). Is it a js symbol? Is it to tell it to search inside colleagues?